router/core.go

```diff
@@ -16,14 +17,12 @@ import (
 	"github.com/piiano/cellotape/router/utils"
 )
 
+var ErrSpecValidation = errors.New("spec validation failed")
+
 func createMainRouterHandler(oa *openapi) (http.Handler, error) {
-	// Customize the error message returned by the kin-openapi library to be more user-friendly.
-	openapi3filter.DefaultOptions.WithCustomSchemaErrorFunc(func(err *openapi3.SchemaError) string {
-		return err.Reason
-	})
 	flatOperations := flattenOperations(oa.group)
 	if err := validateOpenAPIRouter(oa, flatOperations); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%w: %s", ErrSpecValidation, err)
```

Comment:
Following Go1.20 you can use `"%w: %w"`. If you prefer not to, please use `%v` instead of `%s` for errors.